clients/src/main/java/org/apache/kafka/clients/consumer/Consumer.java

```diff
@@ -71,12 +71,12 @@
     /**
      * @see KafkaConsumer#commit(Map, CommitType)
      */
-    public void commit(Map<TopicPartition, Long> offsets, CommitType commitType);
+    public void commit(Map<TopicPartition, OffsetMetadata> offsets, CommitType commitType);
```

Comment:
I've got an even uglier idea. Reverse the order of the last two arguments:
```java
public void commit(Map<TopicPartition, OffsetAndMetadata> offsets, ConsumerCommitCallback callback, CommitType commitType);
```
More seriously, I don't think commitWithMetadata is so bad.